src/java/fr/paris/lutece/plugins/workflow/modules/appointmentants/service/TaskAddAntsAppointment.java

```diff
@@ -0,0 +1,167 @@
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * License 1.0
+ */
+package fr.paris.lutece.plugins.workflow.modules.appointmentants.service;
+
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.servlet.http.HttpServletRequest;
+
+import fr.paris.lutece.api.user.User;
+import fr.paris.lutece.plugins.appointment.business.appointment.Appointment;
+import fr.paris.lutece.plugins.appointment.service.AppointmentService;
+import fr.paris.lutece.plugins.workflow.modules.appointmentants.service.rest.TaskAntsAppointmentRest;
+import fr.paris.lutece.plugins.workflow.modules.appointmentants.service.rest.TaskAntsAppointmentRestConstants;
+import fr.paris.lutece.plugins.workflowcore.business.resource.ResourceHistory;
+import fr.paris.lutece.plugins.workflowcore.service.config.ITaskConfigService;
+import fr.paris.lutece.plugins.workflowcore.service.resource.IResourceHistoryService;
+import fr.paris.lutece.plugins.workflowcore.service.resource.ResourceHistoryService;
+import fr.paris.lutece.plugins.workflowcore.service.task.SimpleTask;
+import fr.paris.lutece.portal.service.i18n.I18nService;
+import fr.paris.lutece.portal.service.util.AppLogService;
+import fr.paris.lutece.portal.service.util.AppPropertiesService;
+import fr.paris.lutece.util.httpaccess.HttpAccessException;
+
+/**
+ * 
+ * Workflow task used to add an appointment in the ANTS' database, through their exposed API
+ * 
+ */
+public class TaskAddAntsAppointment extends SimpleTask
+{
+	public static final String CLASS_NAME = WorkflowAppointmentAntsPlugin.PLUGIN_NAME + "TaskAddAntsAppointment";
+
+	@Inject
+	@Named( ResourceHistoryService.BEAN_SERVICE )
+	private IResourceHistoryService _resourceHistoryService;
+
+	/**
+	 * Task configuration
+	 */
+	@Inject
+	@Named( WorkflowAppointmentAntsPlugin.BEAN_CONFIG )
+	private ITaskConfigService _config;
+
+	@Inject
+	@Named( TaskAntsAppointmentService.BEAN_SERVICE )
+	private ITaskAntsAppointmentService _antsAppointmentService;
+
+	/**
+	 * Title of the task
+	 */
+	private static final String PROPERTY_LABEL_TITLE = "module.workflow.appointmentants.add_appointment.task_title";
+
+	@Override
+	public void processTask( int nIdResourceHistory, HttpServletRequest request, Locale locale, User user )
```

Comment:
I appreciate your considering the use of the latest version of the workflow-core library. Kindly implement the processTaskWithResult method instead of processTask to manage the task's execution status. If the appointment insertion is successful, please return true; otherwise, return false.